rpcserver.go

```diff
@@ -3001,9 +3028,44 @@ func (r *rpcServer) QueryProof(ctx context.Context,
 		"(universeID=%x, leafKey=%x)", universeID,
 		leafKey.UniverseKey())
 
+	// Keep a record of whether the proof type was specified by the client.
+	unspecifiedArgProofType :=
+		universeID.ProofType == universe.ProofTypeUnspecified
+	if unspecifiedArgProofType {
+		// The target proof type was unspecified. Assume that the proof
+		// type is a transfer proof (more transfer proofs exist then
+		// issuance proofs).
+		universeID.ProofType = universe.ProofTypeTransfer
+	}
+
+	// Attempt to fetch the proof from the base universe.
 	proofs, err := r.cfg.BaseUniverse.FetchIssuanceProof(
 		ctx, universeID, leafKey,
 	)
+	// If we were unable to find a proof, and the proof type was originally
+	// unspecified, then we'll try again with the other proof type.
+	if unspecifiedArgProofType &&
+		errors.Is(err, universe.ErrNoUniverseProofFound) {
+
+		// The proof type was originally unspecified by the client. We
+		// will therefore try again with the other proof type.
+		switch universeID.ProofType {
```

Comment:
Ah ok, I see the switcheroo here. 

I do wonder if we should just hide this one layer deeper, so at the layer right above the db where methods like `FetchIssuanceProof` are implemented.